woocommerce/woocommerce
# Fix most of lint warnings

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





There are 30 linter warnings across blocks. This PR aims to fix most of them so there's less noise in the linter output.

[Example job.](https://github.com/woocommerce/woocommerce/actions/runs/10490541656/job/29057579426?pr=48984)

Unfortunately, I don't feel comfortable fixing the remaining three which are related to hooks dependencies in the scope of Cart and Checkout and notices:
```
woocommerce/woocommerce/plugins/woocommerce-blocks/assets/js/base/context/hooks/cart/use-store-cart-coupons.ts
44:4  warning  React Hook useSelect has unnecessary dependencies: 'createErrorNotice' and 'createNotice'. Either exclude them or remove the dependency array  react-hooks/exhaustive-deps

woocommerce/woocommerce/plugins/woocommerce-blocks/assets/js/base/context/hooks/cart/use-store-cart.ts
5:3  warning  React Hook useSelect has missing dependencies: 'isEditor', 'previewCart.coupons', 'previewCart.cross_sells', 'previewCart.fees', 'previewCart.has_calculated_shipping', 'previewCart.items', 'previewCart.items_count', 'previewCart.items_weight', 'previewCart.needs_payment', 'previewCart.needs_shipping', 'previewCart.paymentRequirements', 'previewCart.receiveCart', 'previewCart.receiveCartContents', 'previewCart.shipping_rates', and 'previewCart.totals'. Either include them or remove the dependency array  react-hooks/exhaustive-deps
```



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Only to be tested by developers

1. Go to actions `CI / Lint - @woocommerce/block-library` -> `Lint`
2. Make sure there are only 3 warnings
3. All the jobs (especially tests should pass)



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
